hazelcast/src/main/java/com/hazelcast/internal/util/ExceptionUtil.java

```diff
@@ -194,4 +224,27 @@ public static RuntimeException rethrowAllowInterrupted(final Throwable t) throws
             }
         };
     }
+
+    public static <T extends Throwable> T tryWrapInSameClass(T cause) {
+        Class<? extends Throwable> exceptionClass = cause.getClass();
+        MethodHandle constructor;
+        try {
+            constructor = LOOKUP.findConstructor(exceptionClass, MT_INIT_STRING_THROWABLE);
```

Comment:
We seem to do this lookup for the same exception many times if many exceptions. Do you think that we should cache it so that reflection lookup cost is minimal?

Reply:
Reflection lookup is not costly enough to use caching here.